### PR TITLE
Updating fourchette to work with migrated Heroku pg:backups

### PR DESCRIPTION
**Pull request by wless1**

### CODE REVIEW

File: lib/fourchette/heroku.rb [hunk 1]
```diff
@@ -30,6 +30,12 @@ def client
     @heroku_client ||= PlatformAPI.connect(api_key)
   end
 
+  def legacy_client
+    api_key = ENV['FOURCHETTE_HEROKU_API_KEY']
+    ENV['HEROKU_API_KEY'] = api_key # necessary for Heroku::Auth.password to work
```
Comment (jipiboily) [inline lib/fourchette/heroku.rb]: Why is that necessary? `@non_platform_client` is not even used anywhere in that PR...?

Comment (jipiboily) [inline lib/fourchette/heroku.rb]: Oh, nevermind, it's used... :P

but why is the `ENV['HEROKU_API_KEY']` required is still a valid question.

File: lib/fourchette/pgbackups.rb [hunk 2]
```diff
@@ -7,22 +10,33 @@ def initialize
...
   def copy(from, to)
-    ensure_pgbackups_is_present(from)
-    ensure_pgbackups_is_present(to)
-
     from_url, from_name = pg_details_for(from)
     to_url, to_name = pg_details_for(to)
 
-    @client =  Heroku::Client::Pgbackups.new pgbackup_url(from) + '/api'
-    @client.create_transfer(from_url, from_name, to_url, to_name)
+    raw_attachment = @heroku.legacy_client.get_attachments(to).body[0]
```
Comment (jipiboily) [inline lib/fourchette/pgbackups.rb]: Is that taking the most recent one? What if there is no backup yet?

**Comment by jipiboily** [comment]
Thanks for the PR!

The build is failing.

I added a couple of questions. I will need more time to test it. At first, it seems fine.

@scruti @thinkmorebetter: could you look at it and maybe try it on your apps? Would be awesome!

Thanks

**Comment by wless1** [comment]
No problem. It's still a work in progress, I was planning on fixing the tests, and using it a bit more with our app to iron out any issues (or example, our app needs 2X dynos, so I was hoping on getting that working).

File: lib/fourchette/heroku.rb (hunk 1 above)
Comment (wless1) [inline lib/fourchette/heroku.rb]: I couldn't find support in their platform API for the migrated PG backups, so I was forced to use the ruby client. One of the problems with it is that its Heroku::Client::HerokuPostgresql implicitly uses Heroku::Auth, which tries to read credentials from stdin. Only means I could see of bypassing this behavior is setting this environment variable before it does so.

File: lib/fourchette/pgbackups.rb (hunk 2 above)
Comment (wless1) [inline lib/fourchette/pgbackups.rb]: it's actually doing a pg:copy - so it doesn't rely on a backup at all. Perhaps this file's name should be changed to reflect that?

**Comment by wless1** [comment]
@jipiboily I've fixed the specs - what do you think about merging this?

**Comment by teldio-operations** [comment]
@wless1 Gave this branch a shot and ended up with the following error

```
I, [2015-04-29T19:40:18.580547 #18]  INFO -- : Waiting for database #{to_name} to become available
RestClient::ResourceNotFound: 404 Resource Not Found: {"error":"not found"}
```

Not sure if I am doing something wrong, but the stack trace seems to suggest that it is a result ```wait_for_db``` in ```pgbackups.rb```

File: lib/fourchette/pgbackups.rb [hunk 3]
```diff
@@ -7,42 +10,42 @@ def initialize
...
-    ensure_pgbackups_is_present(to)
-
     from_url, from_name = pg_details_for(from)
     to_url, to_name = pg_details_for(to)
 
-    @client =  Heroku::Client::Pgbackups.new pgbackup_url(from) + '/api'
-    @client.create_transfer(from_url, from_name, to_url, to_name)
-  end
+    raw_attachment = @heroku.legacy_client.get_attachments(to).body[0]
+    attachment = Heroku::Helpers::HerokuPostgresql::Attachment.new raw_attachment
```
Comment (teldio-operations) [inline lib/fourchette/pgbackups.rb]: Getting the 0 indexed attachment does not work for me. We should probably analyze each attachment until we find the attachment with a name starting with ```HEROKU_POSTGRESQL```

File: lib/fourchette/pgbackups.rb [hunk 4]
```diff
@@ -7,42 +10,42 @@ def initialize
...
 
-  private
+    pg_client =  Heroku::Client::HerokuPostgresql.new(attachment)
 
-  def ensure_pgbackups_is_present(heroku_app_name)
-    unless existing_backups?(heroku_app_name)
-      logger.info "Adding pgbackups to #{heroku_app_name}"
-      @heroku.client.addon.create(heroku_app_name, { plan: 'pgbackups' })
-    end
+    logger.info 'Waiting for database #{to_name} to become available'
```
Comment (jipiboily) [inline lib/fourchette/pgbackups.rb]: Can you change the `'` to `"` so `#{to_name}` can do its job please?

**Comment by jipiboily** [comment]
@wless1 sorry it's been so long. What's the status of this? Is this working for you?

**Comment by jipiboily** [comment]
FYI, we just decided to deprecate Fourchette. You can (and probably should) use the Heroku review apps instead which was inspired by Fourchette. 

https://devcenter.heroku.com/articles/github-integration-review-apps

Thanks!

**Comment by jipiboily** [comment]
Closing, feel free to re-open if you think it's worth pursuing.
